activerecord/lib/active_record/associations.rb

```diff
@@ -49,28 +34,21 @@ def initialize(reflection = nil, associated_class = nil)
       end
     end
 
-    class Correction
-      def initialize(error)
-        @error = error
-      end
+    if defined?(DidYouMean::Correctable) && defined?(DidYouMean::Jaro)
+      include DidYouMean::Correctable
 
       def corrections
-        if @error.reflection && @error.associated_class
-          maybe_these = @error.associated_class.reflections.keys
+        if reflection && associated_class
+          maybe_these = associated_class.reflections.keys
 
           maybe_these.sort_by { |n|
-            DidYouMean::Jaro.distance(@error.reflection.options[:inverse_of].to_s, n)
+            DidYouMean::Jaro.distance(reflection.options[:inverse_of].to_s, n)
```

Comment:
Can we use `DidYouMean::SpellChecker` here?

Reply:
Done!